src/realizations/catchment/Tshirt_C_Realization.cpp

```diff
@@ -495,7 +495,13 @@ int Tshirt_C_Realization::run_formulation_for_timesteps(std::vector<double> inpu
     //aorc_forcing_data empty_forcing[num_timesteps];
     aorc_forcing_data empty_forcing[1];
 
-    double* input_as_array = &input_fluxes[0];
+    // Since the input_fluxes param values are in meters per second, this will need to do some conversions to what gets
+    // passed to Tshirt_C's run(), which expects meters per hour.
+    std::vector<double> input_meters_per_hour(input_fluxes.size());
+    for (int i = 0; i < input_fluxes.size(); ++i) {
+        input_meters_per_hour[i] = input_fluxes[i] * 3600;
+    }
+    double* input_as_array = &input_meters_per_hour[0];
```

Comment:
I agree that this needs to be done, but not sure about doing it here.  When I fixed this locally I applied the conversion in `get_response` like this:

```diff
@@ -407,6 +408,8 @@ double Tshirt_C_Realization::get_response(time_step_t t_index, time_step_t t_del
 
     // TODO: this is problematic, because what happens if the wrong t_index is passed?
     double precip = this->forcing.get_next_hourly_precipitation_meters_per_second();
+    //convertp precip from m/s to m/time_step
+    precip = precip * t_delta_s;
     int response_result = run_formulation_for_timestep(precip);
 
     // TODO: check t_index is the next expected time step to be calculated

```

The conversion in `run_formulation_for_timesteps` ASSUMES a `dt` of 3600.  The rest of the formulation works in a `m/timestep` fashion (but is there ever an assumption that that timestep is 1 hr?), so this function is assuming inputs are in m/s, which is probably reasonable, but then hard codes the assumption that they are valid for 3600 seconds.  It works for the intended use right now, but we either need to move the conversion up or pass the delta time down.

Reply:
@hellkite500, this relates to a higher-level issue I think needs to be address, in that `get_response` probably needs to be changed to not use any parameters.  I'll explain a little further down, but the above reviewed code was written with that eventuality in mind.  That matters here because the potential for inconsistent conversion factor for precipitation is only due to bad premise currently built into the code, and thus probably temporary. 

Regarding exactly where the unit conversion logic is, that is primarily for simplicity.  Any calls to `run_formulation_for_timestep` or `get_response` eventually result in a nested called to `run_formulation_for_timesteps`.   All expect to receive or read data in m/s, so converting ahead of this would require either duplicated code or a more significant rewriting.  

As for changing `get_response` ...

At least in some (and perhaps in all) formulations, including this one, it only ever makes sense to run the "next" time step, so `t_index` just causes problems.  Pretty sure this already qualifies as a bug (we get the "next" forcing precipitation value regardless of the `t_index`), and we just don't have other things written that exercise the problematic states.  I suppose we could consider adding an overloaded function that uses it and calculates multiple time steps sequentially, but we need to do something differently.

I also think the time step size should be taken out as a parameter.  I am fairly sure we want to support models requiring a static time step size, and I want to say the C-base Tshirt in particular does this.  Given that, it seems more appropriate not have a time step size in `get_response`, and instead expect formulations to handle whether/how time step size is mutable within their own implementations.  

I'm open to discussing those further, but I wasn't going to include them as part of this, since they seemed out of scope.  They may be a necessary prereq, though.

Reply:
@hellkite500 from our conversation yesterday, I made some changes.  I have left the conversion logic where it was so it only has to be coded in one spot, but the class is now passing the `dt` value to the `run_formulation_*` functions to make sure the appropriate value is available, rather than just something hard coded.